lib/LWP/UserAgent.pm

```diff
@@ -1005,10 +1005,14 @@ sub mirror
             $request->header( 'If-Modified-Since' => HTTP::Date::time2str($mtime) );
         }
     }
-    my $tmpfile = "$file-$$";
+
+    require File::Temp;
+    my ($tmpfh, $tmpfile) = File::Temp::tempfile("$file-XXXXXX");
+    close($tmpfh) or die "Could not close tmpfile '$tmpfile': $!";
```

Comment:
Perhaps instead of this we could teach LWP::Protocol to [dup filehandles](https://perldoc.perl.org/functions/open#Duping-filehandles):
```
diff --git a/lib/LWP/Protocol.pm b/lib/LWP/Protocol.pm
index ef12a40f..a220a487 100644
--- a/lib/LWP/Protocol.pm
+++ b/lib/LWP/Protocol.pm
@@ -110,8 +110,9 @@ sub collect
             if (!defined($arg) || !$response->is_success) {
                 $response->{default_add_content} = 1;
             }
-            elsif (!ref($arg) && length($arg)) {
-                open(my $fh, ">", $arg) or die "Can't write to '$arg': $!";
+            elsif (fileno($arg) || !ref($arg) && length($arg)) {
+                my $mode = fileno($arg) ? '>&' : '>';
+                open(my $fh, $mode, $arg) or die "Can't write to '$arg': $!";
                 binmode($fh);
                 push(@{$response->{handlers}{response_data}}, {
                     callback => sub {
```
and then this could become:
```
my $tmpfile = File::Temp->new("$file-XXXXXX");
```

And then, with the `dup` we no longer have a race condition on the tmpfile name which I think would close #362.

Reply:
I filed a PR for that over in #400